ActiveLabel/ActiveLabel.swift

```diff
@@ -139,22 +148,139 @@ typealias ElementTuple = (range: NSRange, element: ActiveElement, type: ActiveTy
         didSet { textContainer.lineBreakMode = lineBreakMode }
     }
 
+    private func setupLongPressGesture() {
+        self.longPressGesture.minimumPressDuration = 0.5
+        self.longPressGesture.addTarget(self, action: #selector(self.longPress(_:)))
+        self.addGestureRecognizer(self.longPressGesture)
+    }
+    
+    @objc private func longPress(_ sender: UILongPressGestureRecognizer) {
+        let location = sender.location(in: sender.view)
+        guard let element = self.element(at: location) else {
+            return
+        }
+        var link: URL? = nil
+        switch element.element {
+        case .mention(let userHandle):
+            link = URL(string: "@" + userHandle)
+        case .hashtag(let hashtag):
+            link = URL(string: "#" + hashtag)
+        case .url(let originalURL, _):
+            link = URL(string: originalURL)
+        case .custom(_):
+            break
+        }
+        guard let url = link else {
+            return
+        }
+        self.processLink(url, touchPoint: location, sender: sender)
+    }
+
+    private func processLink(_ url: URL, touchPoint: CGPoint, sender: UILongPressGestureRecognizer) {
+        self.delegate?.didLongPressWithURL(url, touchPoint: touchPoint)
+        guard self.copyLinksActive,
+            let rect = self.selectedLinkRectangle(link: url.description, touchPoint: touchPoint),
+            self.lastCopyMenuRect != rect,
+            let responder = sender.view else {
+            return
+        }
+        self.lastCopyMenuRect = rect
+        self.showCopyMenu(rect: rect, responder: responder)
+        self.copyLink = url.description
+    }
+    
+    private func selectedLinkRectangle(link: String, touchPoint: CGPoint) -> CGRect? {
+        guard let text = self.text else {
+            return nil
+        }
+        let linkRanges = self.linkRanges(text: text, link: link)
+        for linkRange in linkRanges {
+            let characterRange = NSRange(linkRange, in: text)
+            let boundRect = boundingRectangle(forCharacterRange: characterRange)
+            let correctBoundRect = CGRect(x: boundRect.minX, y: boundRect.minY + self.heightCorrection, width: boundRect.width, height: boundRect.height)
+            guard !correctBoundRect.contains(touchPoint) else {
```

Comment:
guard is expression for early return from method (invariant check). But here guard fail is used as main flow.